### PR TITLE
Rspec for Task and Comment Controllers/Models

### PR DESCRIPTION
**Pull request by kanatsanan6**
- Change redirect destinations of Controllers
- Add rspec test for
  - Task Model + Controller
  - Comment Model + Controller

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-source 'https://rubygems.org'
+source 'https://www.rubygems.org'
```
Comment (aunsira) [inline Gemfile]: Why?

File: spec/controllers/comments_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    let(:params) { { task_id: task.id, comment: { commenter: 'Test commenter', body: 'Test body' } } }
+    subject { post :create, params: params }
+
+    it { is_expected.to have_http_status(302) }
+    it { is_expected.to redirect_to task_path(assigns(:task)) }
+
+    it 'creates a new comment' do
+      subject
+
+      expect(assigns(:task)).to eq task
```
Comment (aunsira) [inline spec/controllers/comments_controller_spec.rb:21]: You could also have an assertion to make sure resource has been created by 
```ruby
expect(Foo.count).to eq 1
```

File: spec/controllers/comments_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CommentsController, type: :controller do
+  let!(:task) { create(:task) }
+  let!(:comment) { create(:comment, task: task) }
+
+  describe 'POST #create' do
+    let(:params) { { task_id: task.id, comment: { commenter: 'Test commenter', body: 'Test body' } } }
```
Comment (aunsira) [inline spec/controllers/comments_controller_spec.rb]: Please break to block if line is too long.
```ruby
let(:params) do
  {
     task_id: ...
  }
end
```

File: spec/controllers/comments_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    it 'cannot create a new comment' do
+      params[:comment][:commenter] = ''
+      subject
+
+      expect(response).to redirect_to task_path(assigns(:task))
+    end
+  end
+
+  describe 'GET #edit' do
+    let!(:params) { { task_id: task.id, id: comment.id } }
```
Comment (aunsira) [inline spec/controllers/comments_controller_spec.rb]:
```suggestion
    let(:params) { { task_id: task.id, id: comment.id } }
```

File: spec/controllers/comments_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    it 'returns an error' do
+      params[:id] = 'invalid_id'
+
+      expect { subject }.to raise_error(ActiveRecord::RecordNotFound)
+    end
+  end
+
+  describe 'PATCH #update' do
+    let!(:params) do
```
Comment (aunsira) [inline spec/controllers/comments_controller_spec.rb]: This one too.

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ENV['RAILS_ENV'] ||= 'test'
+
+require_relative '../config/environment'
+
+abort('The Rails environment is running in production mode!') if Rails.env.production?
+
+require 'rspec/rails'
+
+
+DatabaseCleaner.strategy = :truncation
```
Comment (aunsira) [inline spec/rails_helper.rb]: Why we have this? (See you already have `require 'support/database_cleaner'`)

File: spec/rails_helper.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+end
+
+begin
+  ActiveRecord::Migration.maintain_test_schema!
+rescue ActiveRecord::PendingMigrationError => e
+  puts e.to_s.strip
+  exit 1
+end
+
+RSpec.configure do |config|
```
Comment (aunsira) [inline spec/rails_helper.rb]: Please decide which file you going to have this RSpec config block.

File: Gemfile (hunk 1 above)
Comment (kanatsanan6) [reply Gemfile]: Yesterday, I had an issue with `bundle install`, and got a suggestion to change the source, but right now it works just fine without this. I will change it back right away.

File: spec/rails_helper.rb (hunk 6 above)
Comment (kanatsanan6) [reply spec/rails_helper.rb]: My bad, I forgot to remove it 😢

File: spec/controllers/comments_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    it 'returns an error' do
+      params[:id] = 'invalid_id'
+
+      expect { subject }.to raise_error(ActiveRecord::RecordNotFound)
+    end
+  end
+
+  describe 'PATCH #update' do
+    let(:params) do
+      { task_id: task.id, id: comment.id, comment: { commenter: 'Updated commenter', body: 'Updated body' } }
```
Comment (aunsira) [inline spec/controllers/comments_controller_spec.rb:58]: Can be better!
```rb
let(:params) do
  {
    id: task2.id,
    task: {
      title: 'updated Title',
      body: 'This is updated body',
      status: 'in_progress'
    }
  }
end
```

**Review by aunsira — APPROVED** [review]

**Pull request merged**